package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,702 @@
+        if not self.is_open:
+            raise IOError("No file open")
+        if self.mode != 'r':
+            raise IOError('File opened in mode: {}. Reading only allow '
+                               'in mode "r"'.format('self.mode'))
+        if self.n_frames == 0:
+            raise IOError("opened empty file. No frames are saved")
+
+        cdef np.ndarray xyz = np.empty((self.natoms, self.ndims), dtype=FLOAT, order='F')
+        cdef np.ndarray unitcell = np.empty(6, dtype=DOUBLE)
+        unitcell[0] = unitcell[2] = unitcell[5] = 0.0;
+        unitcell[4] = unitcell[3] = unitcell[1] = 90.0;
+
+        first_frame = self.current_frame == 0
+        ok = self.c_readframes_helper(xyz[:, 0], xyz[:, 1], xyz[:, 2], unitcell, first_frame)
+        if ok != 0 and ok != -4:
+            raise IOError("Reading DCD header failed: {}".format(DCD_ERRORS[ok]))
+
+        # we couldn't read any more frames.
+        if ok == -4:
+            self.reached_eof = True
+            raise StopIteration
+
+        self.current_frame += 1
+        return DCDFrame(xyz, unitcell)
+
+
+    def readframes(self, start=None, stop=None, step=None, order='fac', indices=None):
+        """
+        read multiple frames at once
+
+        Parameters
+        ----------
+        start : int (optional)
+            starting frame, default to 0
+        stop : int (optional)
+            stop frame, default to ``n_frames``
+        step : int (optional)
+            step between frames read, defaults to 1
+        order : str (optional)
+            give order of returned array with `f`:frames, `a`:atoms, `c`:coordinates
+        indices : array_like (optional)
+            only read selected atoms. In ``None`` read all.
+
+        Returns
+        -------
+        DCDFrame : namedtuple
+            positions are in ``x`` and unitcell in ``unitcell`` attribute of DCDFrame.
+            Here the attributes contain the positions for all frames in the given order
+
+        Notes
+        -----
+        unitcell is read as it from DCD. Post processing depending the program this
+        DCD file was written with is necessary. Have a look at the MDAnalysis DCD reader
+        for possible post processing into a common unitcell data structure.
+
+        Raises
+        ------
+        IOError
+        ValueError
+        """
+        if self.reached_eof:
+            raise IOError('Reached last frame in DCD, seek to 0')
+        if not self.is_open:
+            raise IOError("No file open")
+        if self.mode != 'r':
+            raise IOError('File opened in mode: {}. Reading only allow '
+                               'in mode "r"'.format('self.mode'))
+        if self.n_frames == 0:
+            raise IOError("opened empty file. No frames are saved")
+
+        self.seek(0)
+        # if we only want to iterate backwards flip start and end
+        if start is None and stop is None and step is not None and step < 0:
```

Comment:
I don't know. This code reads the whole file which has nothing to do with the issue as far as I can see.